### DISCUSSION TITLE
2D Keypoints from 3D Pose Estimation

### DISCUSSION THREAD

**Discussion by DiegoHerrera1890** [open-mmlab/mmpose#2940] [Q&A]
Hello dear community.
I am using mmPose to do 3D pose estimation and indeed i got 3D Pose estimation working well.
I want to draw by myself the 2D keypoints in the video frame but what I got from 3D Pose estimations are not for 2D.
Here is the code for 3D pose estimation:

`results = []

inferencer = MMPoseInferencer(pose3d="human3d")
inferencer = MMPoseInferencer(pose3d="motionbert_dstformer-ft-243frm_8xb32-120e_h36m")
inferencer = MMPoseInferencer(
    pose3d='configs/body_3d_keypoint/motionbert/h36m/' \
           'motionbert_dstformer-ft-243frm_8xb32-120e_h36m.py',
    pose3d_weights='https://download.openmmlab.com/mmpose/v1/body_3d_keypoint/' \
                   'pose_lift/h36m/motionbert_ft_h36m-d80af323_20230531.pth', device='cuda'
)
    

result_generator = inferencer(video_path,vis_out_dir='vis_results', show=True) 
for result in result_generator:
    results.append(result)
`
the output for the keypoints looks like:

`keypoints = [[-0.0, 0.0, 0.6706308722496033], [0.09320202469825745, -0.0077973054721951485, 0.6785376071929932],
             [0.10323791205883026, 0.11077322065830231, 0.3332788348197937], [0.11937475949525833, 0.2293485701084137, 0.0],
             [-0.09267321974039078, 0.009579045698046684, 0.6733694076538086], [-0.053059324622154236, 0.12123174965381622, 0.3170395493507385],
             [-0.004531494341790676, 0.2649613618850708, 0.0040634870529174805], [-0.0038011474534869194, -0.04594200849533081, 0.8992895483970642],
             [-0.0058721923269331455, -0.12048772722482681, 1.1318039894104004], [-0.009062865749001503, -0.17221741378307343, 1.237027645111084],
             [-0.007782226428389549, -0.17756949365139008, 1.2795794010162354], [-0.14852721989154816, -0.11693453788757324, 1.1154943704605103],
             [-0.24147924780845642, -0.004160769283771515, 0.9282780885696411], [-0.21335144340991974, -0.12579265236854553, 0.7980388402938843],
             [0.1444392055273056, -0.09810926020145416, 1.1297661066055298], [0.24949683248996735, 0.048139236867427826, 0.9447985887527466],
             [0.26728978753089905, -0.033880215138196945, 0.7824941277503967]]`

Is there a way to use the 2D pose information to draw by myself the keypoint using opencv, or matplotlib?
Thank you so much

**Comment by RenardiAdryan** [comment]
@DiegoHerrera1890 I think you need to use body3d_pose_lifter_demo.py
and use variable called pose_est_results

**Reply by DiegoHerrera1890** [reply]
@RenardiAdryan  Thank you for your anwser. I could run 3D Pose estimation by first using mmPose for 2D pose estimation and MotionBert to do 3D pose estimation. I am not sure if the depth (Z axis) is in meters or cm. and the Z values seems to be weird instead of increasing (because the person is moving towards) the value it stay in the same place(vary only few decimals 0.02).
